packages/dropdowns.next/src/elements/combobox/OptGroup.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { HTMLAttributes, MouseEventHandler, forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import { composeEventHandlers } from '@zendeskgarden/container-utilities';
+import { useText } from '@zendeskgarden/react-theming';
+import { IOptGroupProps } from '../../types';
+import useComboboxContext from '../../context/useComboboxContext';
+import {
+  StyledOptGroup,
+  StyledOption,
+  StyledOptionContent,
+  StyledOptionTypeIcon,
+  StyledSeparator
+} from '../../views';
+
+/**
+ * @extends LiHTMLAttributes<HTMLLIElement>
+ */
+export const OptGroup = forwardRef<HTMLLIElement, IOptGroupProps>(
+  ({ children, content, icon, label, 'aria-label': ariaLabel, onMouseDown, ...props }, ref) => {
+    const { getOptGroupProps, isCompact } = useComboboxContext();
+    /* Prevent listbox collapse */
+    const handleMouseDown: MouseEventHandler = event => event.preventDefault();
+    /* This useText logic ensures that the group has a valid `aria-label`, either
+     * specified directly or provided implicitly by the `label` prop. */
+    const groupAriaLabel = useText(
+      OptGroup,
+      { 'aria-label': ariaLabel },
+      'aria-label',
+      'Group',
+      !label /* condition */
+    );
+    const optGroupProps = getOptGroupProps({
+      'aria-label': (groupAriaLabel || label)!
+    }) as HTMLAttributes<HTMLUListElement>;
+
+    return (
+      <StyledOption
+        isCompact={isCompact}
+        $type="group"
```

Comment:
Same as with `Option`. Since we are still supporting v4 of `styled-components` (`"styled-components": "^4.2.0 || ^5.0.0"`), we should avoid transient props.